### PR TITLE
Add release notes for 0.12.0

### PR DESCRIPTION
**Pull request by cwsteinbach**
Notes for the 0.12.0 release.

### CODE REVIEW

File: site/docs/releases.md [hunk 1]
```diff
@@ -56,8 +56,46 @@ To add a dependency on Iceberg in Maven, add the following to your `pom.xml`:
...
+Apache Iceberg 0.12.0 was released on {{ DATE }}. It consists of {{ NUMCOMMITS }} commits contributed by {{ NUMCOMMITTERS }} over a {{ NUMDAYS }} day period.
+
+High-level features:
+
+* **Core**
+  * Allow Iceberg schemas to specify one or more columns as row identifers [[\#2465](https://github.com/apache/iceberg/pull/2465)]. Note that this is a prerequisite for supporting upserts in Flink.
+  * Added Jdbc [[\#1870](https://github.com/apache/iceberg/pull/1870)] and DynamoDB [[\#2688](https://github.com/apache/iceberg/pull/2688)] Iceberg catalog implementations.
+  * Added predicate pushdown for partitions and files metadata tables [[\#2358](https://github.com/apache/iceberg/pull/2358), [\#2926](https://github.com/apache/iceberg/pull/2926)]
+  * Added a new Compaction API for IcebergV2 and implemented a new compaction strategy [[\#2501](https://github.com/apache/iceberg/pull/2501), [\#2609](https://github.com/apache/iceberg/pull/2609)].
+* **Spark**
```
Comment (rdblue) [inline site/docs/releases.md:70]: I think that `UPDATE` support was also added to Spark extensions.

File: site/docs/releases.md [hunk 2]
```diff
@@ -56,8 +56,46 @@ To add a dependency on Iceberg in Maven, add the following to your `pom.xml`:
...
+  * [[\#2560](https://github.com/apache/iceberg/pull/2560)] added extensions DDL to set identifier fields 
+  * [[\#2660](https://github.com/apache/iceberg/pull/2660)] added support for micro-batch streaming reads for structured streaming in Spark3
+  * [[\#2777](https://github.com/apache/iceberg/pull/2777)] improves the performance of adding a file to a partition
```
Comment (rdblue) [inline site/docs/releases.md:73]: This improves the performance of importing a Hive table by not loading all partitions from Hive and instead pushing the partition filter to the metastore.

File: site/docs/releases.md [hunk 3]
```diff
@@ -56,8 +56,46 @@ To add a dependency on Iceberg in Maven, add the following to your `pom.xml`:
...
+## 0.12.0 Release Notes
 
-## 0.11.1 release notes
+Apache Iceberg 0.12.0 was released on {{ DATE }}. It consists of {{ NUMCOMMITS }} commits contributed by {{ NUMCOMMITTERS }} over a {{ NUMDAYS }} day period.
+
+High-level features:
+
+* **Core**
+  * Allow Iceberg schemas to specify one or more columns as row identifers [[\#2465](https://github.com/apache/iceberg/pull/2465)]. Note that this is a prerequisite for supporting upserts in Flink.
+  * Added Jdbc [[\#1870](https://github.com/apache/iceberg/pull/1870)] and DynamoDB [[\#2688](https://github.com/apache/iceberg/pull/2688)] Iceberg catalog implementations.
```
Comment (rdblue) [inline site/docs/releases.md:67]: JDBC is normally all caps. Also, no need to qualify "catalog" as "Iceberg catalog".

File: site/docs/releases.md [hunk 4]
```diff
@@ -56,8 +56,46 @@ To add a dependency on Iceberg in Maven, add the following to your `pom.xml`:
...
+  * Added predicate pushdown for partitions and files metadata tables [[\#2358](https://github.com/apache/iceberg/pull/2358), [\#2926](https://github.com/apache/iceberg/pull/2926)]
+  * Added a new Compaction API for IcebergV2 and implemented a new compaction strategy [[\#2501](https://github.com/apache/iceberg/pull/2501), [\#2609](https://github.com/apache/iceberg/pull/2609)].
```
Comment (rdblue) [inline site/docs/releases.md:69]: This isn't really user-facing. I think the main update to mention is that there is a new, more flexible compaction action for Spark that can support different strategies, like bin packing and sorting. This also isn't really related to v2.

File: site/docs/releases.md [hunk 5]
```diff
@@ -56,8 +56,46 @@ To add a dependency on Iceberg in Maven, add the following to your `pom.xml`:
...
 </dependencies>
 ```
+## 0.12.0 Release Notes
 
-## 0.11.1 release notes
+Apache Iceberg 0.12.0 was released on {{ DATE }}. It consists of {{ NUMCOMMITS }} commits contributed by {{ NUMCOMMITTERS }} over a {{ NUMDAYS }} day period.
+
+High-level features:
+
+* **Core**
```
Comment (rdblue) [inline site/docs/releases.md:65]: Also added the ability to upgrade to v2 or create a v2 table using the table property `format-verison=2`.

File: site/docs/releases.md [hunk 6]
```diff
@@ -56,8 +56,46 @@ To add a dependency on Iceberg in Maven, add the following to your `pom.xml`:
...
+* The Iceberg Community [voted to approve](https://mail-archives.apache.org/mod_mbox/iceberg-dev/202107.mbox/%3cCAMwmD1-k1gnShK=wQ0PD88it6cg9mY7Y1hKHjDZ7L-jcDzpyZA@mail.gmail.com%3e) version 2 of the Apache Iceberg Format Specification. The differences between version 1 and 2 of the specification are documented [here](https://iceberg.apache.org/spec/#version-2).
+* Bugfixes and stability improvements for NessieCatalog
+* Improvements and fixes for Iceberg's Python library
+* Added a vectorized reader for Apache Arrow [[\#2286](https://github.com/apache/iceberg/pull/2286)] and modified Iceberg's Hive integration so that Hive can leverage its vectorized ORC reader when accessing Iceberg tables [[\#2613](https://github.com/apache/iceberg/pull/2613)].
+* Spark: Add limited support for vectorized reads for Parquet V2 ([\#2749](https://github.com/apache/iceberg/pull/2749))
+
+
+## Past releases
+
+### 0.11.1
```
Comment (rdblue) [inline site/docs/releases.md:98]: This is missing the links section for 0.11.1. See the links for 0.10.0.

File: site/docs/releases.md [hunk 7]
```diff
@@ -56,8 +56,46 @@ To add a dependency on Iceberg in Maven, add the following to your `pom.xml`:
...
+  * [\#2877](https://github.com/apache/iceberg/pull/2877) fixes nested struct pruning in Spark
+  * [\#2757](https://github.com/apache/iceberg/pull/2757) fixes Timestamp without Timezone type support in Spark
+
+Other notable changes:
+
+* The Iceberg Community [voted to approve](https://mail-archives.apache.org/mod_mbox/iceberg-dev/202107.mbox/%3cCAMwmD1-k1gnShK=wQ0PD88it6cg9mY7Y1hKHjDZ7L-jcDzpyZA@mail.gmail.com%3e) version 2 of the Apache Iceberg Format Specification. The differences between version 1 and 2 of the specification are documented [here](https://iceberg.apache.org/spec/#version-2).
+* Bugfixes and stability improvements for NessieCatalog
+* Improvements and fixes for Iceberg's Python library
+* Added a vectorized reader for Apache Arrow [[\#2286](https://github.com/apache/iceberg/pull/2286)] and modified Iceberg's Hive integration so that Hive can leverage its vectorized ORC reader when accessing Iceberg tables [[\#2613](https://github.com/apache/iceberg/pull/2613)].
+* Spark: Add limited support for vectorized reads for Parquet V2 ([\#2749](https://github.com/apache/iceberg/pull/2749))
```
Comment (rdblue) [inline site/docs/releases.md:93]: I don't consider this notable. It does not actually support any Parquet v2 encodings, so it isn't useful to anyone.

File: site/docs/releases.md [hunk 8]
```diff
@@ -25,9 +25,9 @@ The latest version of Iceberg is [{{ versions.iceberg }}](https://github.com/apa
 * [{{ versions.iceberg }} Flink runtime Jar](https://search.maven.org/remotecontent?filepath=org/apache/iceberg/iceberg-flink-runtime/{{ versions.iceberg }}/iceberg-flink-runtime-{{ versions.iceberg }}.jar)
 * [{{ versions.iceberg }} Hive runtime Jar](https://search.maven.org/remotecontent?filepath=org/apache/iceberg/iceberg-hive-runtime/{{ versions.iceberg }}/iceberg-hive-runtime-{{ versions.iceberg }}.jar)
 
-To use Iceberg in Spark, download the runtime Jar and add it to the jars folder of your Spark install. Use iceberg-spark3-runtime for Spark 3, and iceberg-spark-runtime for Spark 2.4.
+To use Iceberg in Spark, download the runtime JAR and add it to the jars folder of your Spark install. Use iceberg-spark3-runtime for Spark 3, and iceberg-spark-runtime for Spark 2.4.
 
-To use Iceberg in Hive, download the iceberg-hive-runtime Jar and add it to Hive using `ADD JAR`.
+To use Iceberg in Hive, download the iceberg-hive-runtime JAR and add it to Hive using `ADD JAR`.
```
Comment (rdblue) [inline site/docs/releases.md:30]: No need for these changes.

Comment (cwsteinbach) [reply site/docs/releases.md:30]: JAR, just like JDBC, is supposed to be capitalized.

File: site/docs/releases.md (hunk 1 above)
Comment (cwsteinbach) [reply site/docs/releases.md:71]: Added

File: site/docs/releases.md (hunk 2 above)
Comment (cwsteinbach) [reply site/docs/releases.md]: Done

File: site/docs/releases.md (hunk 3 above)
Comment (cwsteinbach) [reply site/docs/releases.md]: Done

File: site/docs/releases.md (hunk 4 above)
Comment (cwsteinbach) [reply site/docs/releases.md]: Done

File: site/docs/releases.md (hunk 5 above)
Comment (cwsteinbach) [reply site/docs/releases.md:65]: Done

File: site/docs/releases.md (hunk 6 above)
Comment (cwsteinbach) [reply site/docs/releases.md:99]: Done

File: site/docs/releases.md (hunk 7 above)
Comment (cwsteinbach) [reply site/docs/releases.md]: Removed.

**Comment by hankfanchiu** [comment]
Would we also want to list any dependency updates, e.g. Hive 2.3.8 (#2110), Avro 1.10.1 (#1648), and Parquet 1.12.0 (#2441)?

**Comment by szehon-ho** [comment]
Yea and on that note, I'm not sure if we need to change flink.md to mention Flink version 1.12.1 as well (as it mentions the only supported version is 1.11)?

**Comment by cwsteinbach** [comment]
@hankfanchiu I added the dependency upgrade info to the release notes. Please take a look.
@szehon-ho I think the changes to the Flink docs should be done in a separate patch since they're not part of the release notes.

File: site/docs/releases.md [hunk 9]
```diff
@@ -56,8 +56,58 @@ To add a dependency on Iceberg in Maven, add the following to your `pom.xml`:
...
+High-level features:
+
+* **Core**
+  * Allow Iceberg schemas to specify one or more columns as row identifers [[\#2465](https://github.com/apache/iceberg/pull/2465)]. Note that this is a prerequisite for supporting upserts in Flink.
+  * Added JDBC [[\#1870](https://github.com/apache/iceberg/pull/1870)] and DynamoDB [[\#2688](https://github.com/apache/iceberg/pull/2688)] Iceberg catalog implementations.
+  * Added predicate pushdown for partitions and files metadata tables [[\#2358](https://github.com/apache/iceberg/pull/2358), [\#2926](https://github.com/apache/iceberg/pull/2926)].
+  * Added a new, more flexible compaction action for Spark that can support different strategies such as bin packing and sorting. [[\#2501](https://github.com/apache/iceberg/pull/2501), [\#2609](https://github.com/apache/iceberg/pull/2609)].
+  * Added the ability to upgrade to v2 or create a v2 table using the table property format-version=2  [[\#2887](https://github.com/apache/iceberg/pull/2887)].
+* **Spark**
+  * [[\#2560](https://github.com/apache/iceberg/pull/2560)] added extensions DDL to set identifier fields.
```
Comment (jackye1995) [inline site/docs/releases.md:72]: also added extension DDL to replace partition field, not sure if it's worth mentioning separately (https://github.com/apache/iceberg/pull/2365)

**Review by jackye1995 — APPROVED** [review]
overall looks good to me!

File: site/docs/releases.md (hunk 9 above)
Comment (cwsteinbach) [reply site/docs/releases.md:72]: Added. Thanks for the review @jackye1995

**Pull request merged**